src/robot_manager.cpp

```diff
@@ -780,14 +857,81 @@ try
     }
   }
   res.success = true;
-  return true;  
+  return true;
 }
 catch(resource_registrar::TemotoErrorStack& e)
 {
   res.success = false;
   return true;
 }
 
+bool RobotManager::cancelNavigationGoalCb(RobotCancelNavigationGoal::Request& req, RobotCancelNavigationGoal::Response& res)
+try
+{
+  TEMOTO_INFO_("Cancel Navigation goal request");
+  TEMOTO_DEBUG_STREAM_("Request:\n" << req);
+
+  std::lock_guard<std::mutex> l(mutex_ongoing_navigation_requests_);
+
+  auto navigation_request_it = std::find_if(
+    ongoing_navigation_requests_.begin()
+  , ongoing_navigation_requests_.end()
+  , [&](const auto& ongoing_req)
+  {
+    return req.robot_name == ongoing_req.second.request.robot_name;
+  });
+
+  /*
+   * DECLINE: If priority is low
+   */
+  if (req.priority <= navigation_request_it->second.request.priority)
```

Comment:
refer to our discussion in slack about the lack of request id

Reply:
request_id back.